pkg/apis/mattermost/v1alpha1/clusterinstallation_utils.go

```diff
@@ -217,21 +178,57 @@ func (mattermost *ClusterInstallation) GenerateService(serviceName, selectorName
 					Kind:    "ClusterInstallation",
 				}),
 			},
-			Annotations: svcAnnotations,
+			Annotations: annotations,
 		},
 		Spec: corev1.ServiceSpec{
-			Ports: []corev1.ServicePort{
-				{
-					Port:       8065,
-					TargetPort: intstr.FromString("app"),
-				},
-			},
-			Selector:  ClusterInstallationLabels(selectorName),
-			ClusterIP: corev1.ClusterIPNone,
+			Selector: ClusterInstallationLabels(selectorName),
 		},
 	}
 }
 
+// GenerateService returns the service for Mattermost
+func (mattermost *ClusterInstallation) GenerateService(serviceName, selectorName string) *corev1.Service {
+	// Service has custom annotations
+	annotations := map[string]string{
+		"service.alpha.kubernetes.io/tolerate-unready-endpoints": "true",
+	}
+
+	// Result service
```

Comment:
This may be a bit over-verbose, but not a big deal to me.

Reply:
Still love detailed and verbose comments